dev/Security.AccessControl/Security.AccessControl/GetSecurityDescriptorForAppContainerNames.cpp

```diff
@@ -0,0 +1,116 @@
+﻿#include "pch.h"
+
+#include <wil/resource.h>
+#include <wil/result.h>
+#include "Security.AccessControl.h"
+#include <vector>
+#include <sddl.h>
+#include <AclAPI.h>
+#include <UserEnv.h>
+
+using namespace std;
+
+namespace
+{
+    wil::unique_any_psid GetCurrentProcessSid()
+    {
+        wil::unique_any_psid result;
+        wil::unique_process_heap_ptr<TOKEN_GROUPS> tg;
+        DWORD length = 0;
+        if (!::GetTokenInformation(
+            ::GetCurrentProcessToken(),
+            TokenLogonSid,
+            nullptr,
+            0,
+            &length
+        ))
+        {
+            auto err = ::GetLastError();
+            if (err != ERROR_INSUFFICIENT_BUFFER)
+            {
+                THROW_WIN32(err);
+            }
+            tg.reset(static_cast<TOKEN_GROUPS*>(::HeapAlloc(::GetProcessHeap(), HEAP_ZERO_MEMORY, length)));
+        }
+
+        THROW_IF_WIN32_BOOL_FALSE(::GetTokenInformation(
+            ::GetCurrentProcessToken(),
+            TokenLogonSid,
+            tg.get(),
+            length,
+            &length));
+
+        auto sidLength = ::GetLengthSid(tg->Groups[0].Sid);
+        result.reset(::HeapAlloc(::GetProcessHeap(), 0, sidLength));
```

Comment:
Instead of allocating a new thing, maybe just return "tg" and let the caller root around in it? This is an internal method so nicety isn't necessarily required.